mlflow/pyfunc/__init__.py

```diff
@@ -866,18 +913,42 @@ def spark_udf(spark, model_uri, result_type="double"):
             error_code=INVALID_PARAMETER_VALUE,
         )
 
-    with TempDir() as local_tmpdir:
-        local_model_path = _download_artifact_from_uri(
-            artifact_uri=model_uri, output_path=local_tmpdir.path()
-        )
+    local_model_path = _download_artifact_from_uri(
+        artifact_uri=model_uri, output_path=_get_or_create_model_cache_dir()
+    )
+
+    if env_manager == "local":
         # Assume spark executor python environment is the same with spark driver side.
         _warn_dependency_requirement_mismatches(local_model_path)
+        _logger.warning(
+            "You can set `mlflow.pyfunc.spark_udf` function argument `env_manager` to "
+            "be 'conda' to run inference in restored python environment."
+        )
+    else:
```

Comment:
If `env_manager` is `conda`, can we also print a `_logger.info()` statement indicating that Conda will be used to recreate the software environment for inference and that this may take extra time?